app/javascript/src/Plans/components/ApplicationPlansTable.jsx

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import * as React from 'react'
+import {
+  Button,
+  ButtonVariant,
+  Divider,
+  Form,
+  InputGroup,
+  Pagination as PFPagination,
+  PaginationVariant,
+  TextInput,
+  Toolbar,
+  ToolbarItem
+} from '@patternfly/react-core'
+import {
+  Table,
+  TableHeader,
+  TableBody
+} from '@patternfly/react-table'
+import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon'
+import type { ApplicationPlan, Action } from 'Types'
+
+import './ApplicationPlansTable.scss'
+
+export type Props = {
+  plans: ApplicationPlan[],
+  count: number,
+  searchHref: string,
+  onAction: (action: Action) => void
+}
+
+const ApplicationPlansTable = ({ plans, count, searchHref, onAction }: Props): React.Node => {
+  const tableColumns = [
+    { title: 'Name' },
+    { title: 'Applications' },
+    { title: 'State' }
+  ]
+
+  const tableRows = plans.map(p => ({
+    disableActions: false,
+    cells: [
+      { title: <Button href={p.editPath} component="a" variant="link" isInline>{p.name}</Button> },
+      { title: <Button href={p.applicationsPath} component="a" variant="link" isInline>{p.applications}</Button> },
+      p.state
+    ]
+  }))
+
+  const actionResolver = (_rowData, { rowIndex }: { rowIndex: number }) =>
+    plans[rowIndex].actions.map(a => ({
+      title: a.title,
+      onClick: () => onAction(a)
+    }))
+
+  const url = new URL(window.location.href)
+
+  const selectPerPage = (_event, selectedPerPage) => {
+    url.searchParams.set('per_page', selectedPerPage)
+    url.searchParams.delete('page')
+    window.location.href = url.toString()
+  }
+
+  const goToPage = (page) => {
```

Comment:
I don't know if this way to navigate to a route may have any issue with rails routing.

Reply:
I can verify it's safe. It's the same logic we've been using for pagination in Products/Backends index pages. It follows [our pagination (ruby) module](https://github.com/3scale/porta/blob/master/app/controllers/admin/api/filters/pagination.rb#L0-L1) URL syntax.